### DISCUSSION TITLE
How to load different relationships based on a condition?

### DISCUSSION THREAD

**Discussion by vv-ac** [filamentphp/filament#2238] [Help]
Hi all,

I try to load a BelongsToManySelect field that should either refer to a parent model or to a child model based on the following condition
```
count($record->childentries) > 0
```
In case this condition is true it should generate
```
->relationship('childentries', 'name')
```

if false, it's
```
->relationship('parententries', 'name')
```

All ways I tried to implement it did not work.
I tried to do things like
```
->relationship(function($record) if(count($record->childentries > 0) {...} 
```

Any remark?

Thanks

**Comment by danharrin** [comment]
Obviously this will only work on Edit forms, as `$record` does not yet exist on create

```php
->relationship(fn ($record) => count($record->childentries) > 0 ? 'childentries' : 'parententries', 'name')
```

**Reply by vv-ac** [reply]
And again, I learned sth (from your code).
I am using it on an edit form.

Thank you!

**Reply by vv-ac** [reply]
Just realised, I get the following error from using it:
Object of class Illuminate\Database\Eloquent\Collection could not be converted to int

THe entire code looks like this:
```
->schema([
                BelongsToManyMultiSelect::make('Realted entries')
                             ->relationship(fn ($record) => count($record->childentries > 0) ? 'childentries' : 'parententries', 'name')
                            ]),
```
Well, $record->subentries likely is meant here with "Collection", so the error message means, it's not possible to count those in the form of int? On the other hand, I use that count condition a couple of times and it seems to work just fine.

**Reply by danharrin** [reply]
should be `count($record->childentries) > 0` probably
